File: algorithms/Make Two Arrays Equal by Reversing Sub-arrays/Make Two Arrays Equal by Reversing Sub-arrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        int arr1[1001]={0};
        int arr2[1001]={0};
        
        for(int i =0 ; i<target.size(); i++)
        {
            arr1[arr[i]]++;
            arr2[target[i]]++;
        }
        
        for(int i =0 ;i<=1000;i++)
        {
            if(arr1[i]!=arr2[i])
                return false;
        }
        return true;
    }
};

File: scripts/algorithms/M/Make Two Arrays Equal by Reversing Sub-arrays/Make Two Arrays Equal by Reversing Sub-arrays.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        HashMap<Integer,Integer>hm1=new HashMap();
        for(int i: arr){
            if(hm1.containsKey(i))
                hm1.put(i,hm1.get(i)+1);
            else
                hm1.put(i,1);
        }
        for(int i: target){
            if(hm1.containsKey(i)){
                hm1.put(i,hm1.getOrDefault(i,0)-1);
                if(hm1.get(i)==0)
                    hm1.remove(i);
            }
            else
                return false;
            
        }
        if(hm1.size()==0)
            return true;
        
        return false;
    }
}
